azurelinuxagent/ga/update.py

```diff
@@ -648,58 +645,48 @@ def _upgrade_available(self, base_version=CURRENT_VERSION):
         self.last_attempt_time = now
         protocol = self.protocol_util.get_protocol()
 
-        for update_goal_state in [False, True]:
-            try:
-                if update_goal_state:
-                    protocol.update_goal_state(forced=True)
-
-                manifest_list, etag = protocol.get_vmagent_manifests()
-
-                manifests = [m for m in manifest_list.vmAgentManifests \
-                                if m.family == family and \
-                                    len(m.versionsManifestUris) > 0]
-                if len(manifests) == 0:
-                    logger.verbose(u"Incarnation {0} has no {1} agent updates",
-                                    etag, family)
-                    return False
-
-                pkg_list = protocol.get_vmagent_pkgs(manifests[0])
-
-                # Set the agents to those available for download at least as
-                # current as the existing agent and remove from disk any agent
-                # no longer reported to the VM.
-                # Note:
-                #  The code leaves on disk available, but blacklisted, agents
-                #  so as to preserve the state. Otherwise, those agents could be
-                #  again downloaded and inappropriately retried.
-                host = self._get_host_plugin(protocol=protocol)
-                self._set_agents([GuestAgent(pkg=pkg, host=host) \
-                                     for pkg in pkg_list.versions])
-
-                self._purge_agents()
-                self._filter_blacklisted_agents()
-
-                # Return True if current agent is no longer available or an
-                # agent with a higher version number is available
-                return not self._is_version_eligible(base_version) \
-                    or (len(self.agents) > 0 \
-                        and self.agents[0].version > base_version)
-
-            except Exception as e:
-                if isinstance(e, ResourceGoneError):
-                    continue
+        try:
+            manifest_list, etag = protocol.get_vmagent_manifests()
```

Comment:
now we always call update_goal_state (from get_vmagent_manifests)... let's double check whether that can have any undesired side effects